include/EVT/io/I2C.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef _EVT_I2C_
+#define _EVT_I2C_
+
+#include <stdint.h>
+
+namespace EVT::core::IO {
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+enum class Pin;
+
+class I2C {
+public:
+    /**
+     * Make an instance of an I2C interface that will use the given pins
+     * for clock and data lines.
+     *
+     * @param sclPin The clock pin
+     * @param sdaPin The data pin
+     */
+    I2C(Pin sclPin, Pin sdaPin);
+
+    /**
+     * Write a single byte out over I2C.
+     *
+     * @param addr The 7 bit unshifted I2C address to write to
+     * @param byte The value to write over I2C.
+     */
+    virtual void write(uint8_t addr, uint8_t byte) = 0;
+
+    /**
+     * Read a single byte back from the I2C bus.
+     *
+     * @param addr The 7 bit unshifted I2C address to read from.
+     * @return The byte read from the address.
+     */
+    virtual uint8_t read(uint8_t addr) = 0;
+
+    /**
+     * Write out multiple bytes over I2C. Each byte will be written one by
+     * one.
+     *
+     * @param addr The 7 bit unshifted I2C address to write to.
+     * @param bytes The bytes to write out over I2C.
+     * @param length The number of bytes to write out.
+     */
+    void write(uint8_t addr, uint8_t* bytes, uint8_t length);
+
+    /**
+     * Read multiple bytes from an I2C device.
+     *
+     * @param addr The 7 bit unshifted I2C address to write to
+     * @param bytes The buffer to fill with the read in bytes.
+     * @param length The number of bytes to read.
+     */
+    void read(uint8_t addr, uint8_t* bytes, uint8_t length);
+
+    /**
+     * Read a value from a register that has an 8 bit address and 8 bit value
+     *
+     * @param addr The 7 bit unshifted I2C address to read from.
+     * @param reg The 8 bit register
+     * @return The 8 bit value of the register.
+     */
+    uint8_t readReg(uint8_t addr, uint8_t reg);
```

Comment:
readReg but no writeReg?

Reply:
Yeah there should be a write reg, I'll add that now